File: Array/wuchao/array/7-15 寻找第一个唯一字符的下标/FindUniqChar.java
package com.slow.lib.practice.array;

import java.util.HashMap;
import java.util.LinkedHashMap;

/**
 * @Author wuchao
 * @Date 2020/7/15-10:01 AM
 * @description
 * @email 329187218@qq.com
 * @see
 */
public class FindUniqChar {
    public int firstUniqChar(String s) {
        int[] count = new int[26];
        for(int i=0;i<s.length();i++){
            count[s.charAt(i) - 'a']++;
        }
        for(int i=0;i<s.length();i++){
            if(count[s.charAt(i) - 'a'] == 1){
                return i;
            }
        }
        return -1;
    }
}

File: Array/wuchao/array/7-15 寻找第一个唯一字符的下标/FindUniqChar.kt
package com.slow.lib.practice.array

/**
 * @Author wuchao
 * @Date 2020/7/15-10:28 AM
 * @see
 * @description
 * @email 329187218@qq.com
 */
class FindUniqChar {
    fun firstUniqChar(s: String): Int {
        val count = IntArray(26)
        for(index in s.indices){
            count[s[index] - 'a']++
        }
        for(index in s.indices){
            if(count[s[index] - 'a'] == 1){
                return index
            }
        }
        return -1
    }
}